src/main/java/sg/ncl/MainController.java

```diff
@@ -1349,8 +1376,10 @@ public String validateContributeData(@ModelAttribute("dataset") Dataset dataset,
 			catch (Exception e) {
 				redirectAttributes.addFlashAttribute("message",
 						"You failed to upload " + file.getOriginalFilename() + " => " + e.getMessage());
-			}
-		}
+			} finally {
+                stream.close();
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'stream' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2259)